### PR TITLE
Stretch pt whatever

### PR DESCRIPTION
**Pull request by AshenPheonix**

### CODE REVIEW

**Comment by AshenPheonix** [comment]
Just realized i missed a button

File: src/components/TodoComponents/TodoForm.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react'
+
+export default class Drawer extends Component {
+    constructor(props) {
+        super(props)
+        
+        this.state={
+            name:''
+        }
```
Comment (OmarSalah95) [inline src/components/TodoComponents/TodoForm.js:9]: The goal of this application was to centralize state in your main app component and pass down anything the children components need through prop drilling.

File: src/components/TodoComponents/TodoForm.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        super(props)
+        
+        this.state={
+            name:''
+        }
+
+        this.handleSubmit=this.handleSubmit.bind(this)
+        this.formEdit=this.formEdit.bind(this)
+        this.clear=this.clear.bind(this)
+    }
```
Comment (OmarSalah95) [inline src/components/TodoComponents/TodoForm.js:14]: Binding wouldn't be an issue with prop drilling to pull down the functions from the App component where they modify the state.

File: src/components/TodoComponents/TodoList.js [hunk 3]
```diff
@@ -1,2 +1,45 @@
...
+import TodoForm from './TodoForm'
+import './Todo.css'
+
+export default class TodoList extends Component {
+    constructor(props){
+        super(props)
+        this.checkmark=this.checkmark.bind(this)
+        this.add=this.add.bind(this)
+        this.clear=this.clear.bind(this)
+    }
```
Comment (OmarSalah95) [inline src/components/TodoComponents/TodoList.js:14]: Same note as before.

I like the separation of concerns though, especially on bigger projects but this is to guide students through prop drilling, and recognizing the value of a centralized state and single source of truth

**Review by OmarSalah95 — APPROVED** [review]
Good job getting the app up and working. I did note that the readme basically asked use centralized state(in so many words) as part of the project. 

I do like the separation of concerns if this were a bigger project it would make more sense as well to spread state across multiple `hub` components and prop drill what is needed from there. These however are geared to be more difficult 1 for practice and 2 so that when students reach the later modules and lessons they see the value of doing things using other patterns, and methods.

Outside of that great job!

**Pull request merged**